client/src/components/Form/Elements/FormSelectMany/worker/selectManyMain.ts

```diff
@@ -43,6 +43,16 @@ export function main(options: UnwrapNestedRefs<UseSelectManyOptions>): UnwrapNes
         }
     }
 
+    // In case maintainSelectionOrder is enabled, we need to maintain the order from selectedValues
+    if (options.maintainSelectionOrder) {
+        const selectedValuesArray = Array.from(selectedValues);
+        selectedOptionsFiltered.sort((a, b) => {
+            const aIndex = selectedValuesArray.findIndex((v) => v === stringifyObject(a.value));
+            const bIndex = selectedValuesArray.findIndex((v) => v === stringifyObject(b.value));
```

Comment:
Slight optimization.
`stringifyObject` does not need to be re-run for every selected value.

```suggestion
            const aAsString = stringifyObject(a.value);
            const bAsString = stringifyObject(b.value);

            const aIndex = selectedValuesArray.findIndex((v) => v === aAsString);
            const bIndex = selectedValuesArray.findIndex((v) => v === bAsString);
```